File: lib/pane/layout/templates/split_vertical.ex
defmodule Pane.Layout.Templates.SplitVertical do
  @moduledoc """
  A layout with two vertical panes (50/50 split by default).

  Pane names:
  - :left
  - :right
  """

  @behaviour Pane.Layout.Template

  @impl true
  def pane_names, do: [:left, :right]

  @impl true
  def apply(window_target, cwd, _options) do
    # First pane is created by default with the window, so we just need to reference it
    left_target = "#{window_target}.0"
    
    # Split window horizontally (side by side)
    # Note: In tmux, -h means horizontal split (left/right)
    right_cmd = Pane.Tmux.Pane.split(
      direction: :horizontal, 
      percent: 50, 
      target: left_target,
      cwd: cwd
    )
    right_target = "#{window_target}.1"
    
    # Select the left pane again to make it active
    select_left_cmd = Pane.Tmux.Pane.select(left_target)
    
    # Build the commands list
    commands = [right_cmd, select_left_cmd]
    
    # Return commands and a map of pane positions to targets
    {
      commands,
      %{
        left: left_target,
        right: right_target
      }
    }
  end
end

File: nodejs/src/lib/templates/splitvertical.js
/**
 * A layout with two equal vertical panes side by side.
 * 
 * Pane names:
 * - left
 * - right
 */
const { TmuxPane } = require('../tmux');

/**
 * Returns the list of pane position names defined by this template.
 * 
 * @returns {string[]} Array of pane names
 */
function paneNames() {
  return ['left', 'right'];
}

/**
 * Creates the pane structure within a window.
 * 
 * @param {string} windowTarget - The target identifier for the window
 * @param {string} cwd - The current working directory 
 * @param {Object} options - Additional options
 * @returns {Object} Commands and pane targets
 */
function apply(windowTarget, cwd, options = {}) {
  // First pane is created by default with the window
  const leftTarget = `${windowTarget}.0`;
  
  // Create a right pane that's 50% of the window width
  const splitCommand = TmuxPane.split({ 
    target: leftTarget, 
    horizontal: true, 
    percentage: 50,
    cwd
  });
  
  // The new pane becomes pane 1
  const rightTarget = `${windowTarget}.1`;
  
  // Return commands and pane targets
  return {
    commands: [splitCommand],
    paneTargets: {
      left: leftTarget,
      right: rightTarget
    }
  };
}

module.exports = {
  paneNames,
  apply
};